quill/quill.d.ts

```diff
@@ -1,118 +1,85 @@
-    formatLine(start: number, end: number, formats: any, source: string): void;
-
-
-    insertEmbed(index: number, type: string, url: string): void;
-    insertEmbed(index: number, type: string, url: string, source: string): void;
-
-    updateContents(delta: DeltaStatic): void;
-
-    setContents(delta: DeltaStatic): void;
-
-    setHTML(html: string): void;
-
-    setText(text: string): void;
-
-    getSelection(): RangeStatic;
-
-    setSelection(start: number, end: number): void;
-    setSelection(start: number, end: number, source: string): void;
-    setSelection(range: RangeStatic): void;
-    setSelection(range: RangeStatic, source: string): void;
-
-    prepareFormat(format: string, value: boolean): void;
-
-    focus(): void;
-
-    getBounds(index: number): any;
-
-    registerModule(name: string, callback: (quill: QuillStatic, options: any) => {}): any;
-
-    addModule(name: string, options: any): any;
-
-    getModule(name: string): any;
-
-    onModuleLoad(name: string, callback: (input: any) => {}): void;
-
-    addFormat(name: string, config: any): void;
-
-    addContainer(cssClass: string, before?: number): HTMLDivElement;
-}
-
-declare module "Range"
-{
-	var Range: RangeStatic;
-	export = Range;
-}
-
-declare var Delta: DeltaStatic;
-
-declare module "Delta"{
-	export = Delta;
+declare namespace QuillJS {
+
+
+    export interface BoundsStatic {
+        left: number,
+        top: number,
+        height: number,
+        width: number
+    }
+
+    export interface DeltaStatic {
+        ops?: Array<any>;
+        retain?: any,
+        delete?: any,
+        insert?: any,
+        attributes?: any
+    }
+
+    export interface RangeStatic {
+        new (): RangeStatic;
+        index: number;
+        length: number;
+    }
+
+    export interface QuillStatic {
+        new (selector: string, options?: Object): QuillStatic;
+
+        deleteText(start: number, end: number, source?: string): void;
+        disable(): void;
+        enable(enabled?: boolean): void;
+        getContents(start?: number, end?: number): DeltaStatic;
+        getLength(): number;
+        getText(start?: number, end?: number): string;
+        insertEmbed(index: number, type: string, url: string, source?: string): void;
+        insertText(index: number, text: string, source?: string): void;
+        insertText(index: number, text: string, format: string, value: string, source?: string): void;
+        insertText(index: number, text: string, formats: any, source?: string): void;
+        pasteHTML(): string;
+        setContents(delta: DeltaStatic, source?: string): void;
+        setText(text: string, source?: string): void;
+        update(source?: string): void;
+        updateContents(delta: DeltaStatic, source?: string): void;
+
+        format(name: string, value: any, source?: string): void;
+        formatLine(index: number, length: number, source?: string): void;
```

Comment:
Learning fast here... :-)... 
While the union is applicable (now I know), ```source``` has a default value as well (defaults to 'api' most of the times, but that's implementation details?). So this syntax should be fine right?

```
formatLine(index: number, length: number, source?: "api"|"user"|"silent"): void;
```

e.g.
```
    quillEditor.formatLine(1, 3); //source defaults to 'api' in this case
```
and
```
    var quillEditor = new Quill('#editor');
    quillEditor.formatLine(1, 3, {
            'align': 'right',
            'bold': false,
        });
```

Still going through 'Advanced Types' section in handbook to figure out if ```{ [String]: any }``` can be done inline or I need a type declaration for it :D . Most of my inline attempts are generating build errors so far!